[Bug]: CONSENSUS failure when creating a referral set and referral set not created

### Problem encountered

CONSENSUS failure when creating a referral set and the referral set is not created created.

`2023-09-11T15:41:33.630Z	ERROR	tendermint	consensus/state.go:732	CONSENSUS FAILURE!!!	{"err": "unsupported command: team:{name:\"Team1\" team_url:\"\" avatar_url:\"\"}", "stack": "goroutine 213 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x65\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine.func2()\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:732 +0x4c\npanic({0x48c79e0, 0xc004c6ee70})\n\t/usr/local/go/src/runtime/panic.go:884 +0x213\ncode.vegaprotocol.io/vega/core/events.(*TransactionResult).setTx(0xc004c6edc0?, {0x4d98d60?, 0xc004c3ea80?})\n\t/jenkins/workspace/common/system-tests-wrapper/vega/core/events/transaction_result.go:159 +0xbc5\ncode.vegaprotocol.io/vega/core/events.NewTransactionResultEventFailure({0x599d4f0?, 0xc004e10660?}, {0xc004e0b240, 0x40}, {0xc004c0adc0, 0x40}, {0x5976440, 0xc004c6edb0}, {0x4d98d60, 0xc004c3ea80})\n\t/jenkins/workspace/common/system-tests-wrapper/vega/core/events/transaction_result.go:65 +0x1da\ncode.vegaprotocol.io/vega/core/processor.(*App).SendTransactionResult.func1({0x599d4f0, 0xc004e10660}, {0x59bfd90, 0xc004c3eb40})\n\t/jenkins/workspace/common/system-tests-wrapper/vega/core/processor/abci.go:532 +0x356\ncode.vegaprotocol.io/vega/core/blockchain/abci.(*App).DeliverTx(0xc001576840, {{0xc00472f2c0, 0x140, 0x140}})\n\t/jenkins/workspace/common/system-tests-wrapper/vega/core/blockchain/abci/abci.go:140 +0x762\ncode.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).DeliverTx(0xc004ccc440?, {{0xc00472f2c0?, 0x20?, 0xc004ccc460?}})\n\t/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/app_wrapper.go:37 +0x6c\ngithub.com/tendermint/tendermint/abci/client.(*localClient).DeliverTxAsync(0xc000e4e120, {{0xc00472f2c0?, 0x0?, 0xc000e4e120?}})\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/abci/client/local_client.go:93 +0x105\ngithub.com/tendermint/tendermint/proxy.(*appConnConsensus).DeliverTxAsync(0xc004e01bc0?, {{0xc00472f2c0?, 0x20?, 0xb?}})\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/proxy/app_conn.go:85 +0x26\ngithub.com/tendermint/tendermint/state.execBlockOnProxyApp({0x599b490?, 0xc00029b1f8}, {0x59af2c0, 0xc001570cd0}, 0xc0025334a0, {0x59c0a38, 0xc00029d0e0}, 0x46?)\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/state/execution.go:320 +0x6d9\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x1}, {0xc001559a30, 0x7}}, {0xc001559a40, 0xb}, 0x1, 0x46, {{0xc004d363a0, ...}, ...}, ...}, ...)\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/state/execution.go:140 +0x171\ngithub.com/tendermint/tendermint/consensus.(*State).finalizeCommit(0xc0003c1880, 0x47)\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:1661 +0xafd\ngithub.com/tendermint/tendermint/consensus.(*State).tryFinalizeCommit(0xc0003c1880, 0x47)\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:1570 +0x2ff\ngithub.com/tendermint/tendermint/consensus.(*State).enterCommit.func1()\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:1505 +0xaa\ngithub.com/tendermint/tendermint/consensus.(*State).enterCommit(0xc0003c1880, 0x47, 0x0)\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:1543 +0xccf\ngithub.com/tendermint/tendermint/consensus.(*State).addVote(0xc0003c1880, 0xc004c51d60, {0xc001436b10, 0x28})\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:2165 +0x18dc\ngithub.com/tendermint/tendermint/consensus.(*State).tryAddVote(0xc0003c1880, 0xc004c51d60, {0xc001436b10?, 0xc003c23c08?})\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:1963 +0x2c\ngithub.com/tendermint/tendermint/consensus.(*State).handleMsg(0xc0003c1880, {{0x59792a0, 0xc00400f6a8}, {0xc001436b10, 0x28}})\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:861 +0x3ff\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine(0xc0003c1880, 0x0)\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:768 +0x3f0\ncreated by github.com/tendermint/tendermint/consensus.(*State).OnStart\n\t/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/state.go:379 +0x12d\n"}`

Also the soak tests failed and observed the below error in the soak test logs

```
panic: unsupported command: team:{name:"Team1" team_url:"" avatar_url:""} [recovered]
	panic: unsupported command: team:{name:"Team1" team_url:"" avatar_url:""}

goroutine 101 [running]:
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*Command).Run.func1.1()
	/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/node.go:154 +0x59
panic({0x48c79e0, 0xc00407d690})
	/usr/local/go/src/runtime/panic.go:884 +0x213
code.vegaprotocol.io/vega/core/events.(*TransactionResult).setTx(0xc00407d5f0?, {0x4d98d60?, 0xc004427f40?})
	/jenkins/workspace/common/system-tests-wrapper/vega/core/events/transaction_result.go:159 +0xbc5
code.vegaprotocol.io/vega/core/events.NewTransactionResultEventFailure({0x599d4f0?, 0xc0044529f0?}, {0xc004449200, 0x40}, {0xc004448d00, 0x40}, {0x5976440, 0xc00407d5e0}, {0x4d98d60, 0xc004427f40})
	/jenkins/workspace/common/system-tests-wrapper/vega/core/events/transaction_result.go:65 +0x1da
code.vegaprotocol.io/vega/core/processor.(*App).SendTransactionResult.func1({0x599d4f0, 0xc0044529f0}, {0x59bfd90, 0xc004460000})
	/jenkins/workspace/common/system-tests-wrapper/vega/core/processor/abci.go:532 +0x356
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).DeliverTx(0xc0005b0160, {{0xc00441e140, 0x140, 0x140}})
	/jenkins/workspace/common/system-tests-wrapper/vega/core/blockchain/abci/abci.go:140 +0x762
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).DeliverTx(0xc0044412a0?, {{0xc00441e140?, 0x20?, 0xc0044412c0?}})
	/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/app_wrapper.go:37 +0x6c
github.com/tendermint/tendermint/abci/client.(*localClient).DeliverTxAsync(0xc000f11560, {{0xc00441e140?, 0x0?, 0xc000f11560?}})
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/abci/client/local_client.go:93 +0x105
github.com/tendermint/tendermint/proxy.(*appConnConsensus).DeliverTxAsync(0xc004441fe0?, {{0xc00441e140?, 0x20?, 0xb?}})
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/proxy/app_conn.go:85 +0x26
github.com/tendermint/tendermint/state.execBlockOnProxyApp({0x599b490?, 0xc000728d70}, {0x59af2c0, 0xc00107cc90}, 0xc004412b40, {0x59c0a38, 0xc00114b6c8}, 0x46?)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/state/execution.go:320 +0x6d9
github.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x1}, {0xc0010b2128, 0x7}}, {0xc0010b2140, 0xb}, 0x1, 0x46, {{0xc00108c620, ...}, ...}, ...}, ...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/state/execution.go:140 +0x171
github.com/tendermint/tendermint/consensus.(*Handshaker).replayBlock(_, {{{0xb, 0x1}, {0xc0010b2128, 0x7}}, {0xc0010b2140, 0xb}, 0x1, 0x46, {{0xc00108c620, ...}, ...}, ...}, ...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/replay.go:503 +0x23c
github.com/tendermint/tendermint/consensus.(*Handshaker).replayBlocks(_, {{{0xb, 0x1}, {0xc0010b2128, 0x7}}, {0xc0010b2140, 0xb}, 0x1, 0x46, {{0xc00108c620, ...}, ...}, ...}, ...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/replay.go:481 +0x2fa
github.com/tendermint/tendermint/consensus.(*Handshaker).ReplayBlocks(_, {{{0xb, 0x1}, {0xc0010b2128, 0x7}}, {0xc0010b2140, 0xb}, 0x1, 0x46, {{0xc00108c620, ...}, ...}, ...}, ...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/replay.go:408 +0x8bc
github.com/tendermint/tendermint/consensus.(*Handshaker).Handshake(0xc0035f6d80, {0x59c76e0, 0xc0010ba680})
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/consensus/replay.go:268 +0x3d4
github.com/tendermint/tendermint/node.doHandshake({_, _}, {{{0xb, 0x1}, {0xc0010b2128, 0x7}}, {0xc0010b2140, 0xb}, 0x1, 0x46, ...}, ...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/node/node.go:329 +0x1b8
github.com/tendermint/tendermint/node.NewNode(0xc001521040, {0x5995be0, 0xc0002ab900}, 0xc000ff7420, {0x5979400, 0xc00114aff0}, 0x6f4373656b617453?, 0x6e6f697463656e6e?, 0xc000ff7480, {0x599b490, ...}, ...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/node/node.go:779 +0x597
code.vegaprotocol.io/vega/core/blockchain/abci.NewTmNode({{0x0}, 0x1, 0x1, 0x0, 0x0, {0xc00069b5a0, 0xa}, {{0x0}, {0xc0005d6c90, 0x15}}, ...}, ...)
	/jenkins/workspace/common/system-tests-wrapper/vega/core/blockchain/abci/tm_node.go:83 +0x725
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*Command).startABCI(0xc0010ccd80, 0x0?, {0x59c1240, 0xc0016e4e20}, {0x7fff0a5b50e9, 0x53}, {0x0?, 0x0?}, {0x0, 0x0})
	/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/node.go:426 +0x1b6
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*Command).startBlockchain(0xc0010ccd80, 0x0?, {0x7fff0a5b50e9?, 0x0?}, {0x0, 0x0}, {0x0, 0x0})
	/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/node.go:313 +0x3ab
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*Command).Run.func1()
	/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/node.go:157 +0x99
created by code.vegaprotocol.io/vega/cmd/vega/commands/node.(*Command).Run
	/jenkins/workspace/common/system-tests-wrapper/vega/cmd/vega/commands/node/node.go:148 +0x74f
```

### Observed behaviour

CONSENSUS failure when creating a referral set and the referral set is not created created.

Logs:

**Consensus Failure:**

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/7181/artifact/testnet/logs/testnet-nodeset-validators-0-validator/vega-validator-0.stdout-2023-09-11T15%3A40%3A41Z.log/*view*/

**Soak Test Failure:**

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/7181/artifact/soak-test/err-node-0.log/*view*/

### Expected behaviour

No consensus failiure and the referral set is successfully created.

### Steps to reproduce

```bash
Run the below system test:

System Test Name - test_create_update_join_referral_set
System Test Branch - feat/teams-tests
```


### Software version

vega - develop

### Failing test

test_create_update_join_referral_set

### Jenkins run

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/7181/

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/core/events/transaction_result.go
+++ b/core/events/transaction_result.go
@@ -155,6 +155,19 @@ func (t *TransactionResult) setTx(tx interface{}) *TransactionResult {
 		t.evt.Transaction = &eventspb.TransactionResult_StopOrderCancellation{
 			StopOrderCancellation: tv,
 		}
+	case *commandspb.CreateReferralSet:
+		t.evt.Transaction = &eventspb.TransactionResult_CreateReferralSet{
+			CreateReferralSet: tv,
+		}
+	case *commandspb.UpdateReferralSet:
+		t.evt.Transaction = &eventspb.TransactionResult_UpdateReferralSet{
+			UpdateReferralSet: tv,
+		}
+	case *commandspb.ApplyReferralCode:
+		t.evt.Transaction = &eventspb.TransactionResult_ApplyReferralCode{
+			ApplyReferralCode: tv,
+		}
+
 	default:
 		panic(fmt.Sprintf("unsupported command: %v", tv))
 	}

